craft_parts/overlays/overlay_manager.py

```diff
@@ -127,7 +129,8 @@ def refresh_packages_list(self) -> None:
 
         mount_dir = self._project_info.overlay_mount_dir
         # Ensure we always run refresh_packages_list by resetting the cache
-        packages.Repository.refresh_packages_list.cache_clear()
+        if hasattr(packages.Repository.refresh_packages_list, "cache_clear"):
```

Comment:
Possibly more of a question for @cmatsuoka, but is there any reason to not make the abstract `BaseRepository` have an `lru_cache`?

Reply:
The `packages` package code was copied from Snapcraft and it contains many areas that could be improved (or fully refactored). I can't say much about this specific case without verifying the code in detail.

Reply:
In that case this can be a mystery for another day.